ext/openssl/tests/bug38261.phpt

```diff
@@ -15,19 +15,31 @@ class test {
 $t = new test;
 
 var_dump(openssl_x509_parse("foo"));
-var_dump(openssl_x509_parse($t));
-var_dump(openssl_x509_parse(array()));
+
+try {
+    var_dump(openssl_x509_parse($t));
+} catch (TypeError $e) {
+    echo $e->getMessage(), "\n";
+}
+
+try {
+    openssl_x509_parse([]);
+} catch (TypeError $e) {
+    echo $e->getMessage(), "\n";
+}
+
 var_dump(openssl_x509_parse($cert));
+
 try {
-    var_dump(openssl_x509_parse(new stdClass));
-} catch (Error $e) {
+    openssl_x509_parse(new stdClass);
+} catch (TypeError $e) {
     echo $e->getMessage(), "\n";
 }
 
 ?>
 --EXPECT--
 bool(false)
+openssl_x509_parse(): Argument #1 ($x509) must be of type X509Certificate|string, test given
```

Comment:
I think this is a problem with the `Z_PARAM_STR_OR_OBJ_OF_CLASS` macro. If an object is not a X509Certificate then ZPP fails, rather than trying to cast the object to a string. I think it's the right time to fix it. :)